src/tsd.c

```diff
@@ -122,20 +144,17 @@ tsd_cleanup(void *arg) {
 	case tsd_state_nominal:
 	case tsd_state_nominal_slow:
 	case tsd_state_reincarnated:
+		prof_tdata_cleanup(tsd);
+		iarena_cleanup(tsd);
+		arena_cleanup(tsd);
+		arenas_tdata_cleanup(tsd);
+		tcache_cleanup(tsd);
+		witnesses_cleanup(tsd);
```

Comment:
Oh I mean to put these *_cleanup functions in a separate static function, something like tsd_members_cleanup